OneSignalSDK/onesignal/src/main/java/com/onesignal/OneSignal.java

```diff
@@ -815,25 +816,27 @@ private static void handleAmazonPurchase() {
    private static void doSessionInit() {
       // Check session time to determine whether to start a new session or not
       if (isPastOnSessionTime()) {
-         OneSignal.onesignalLog(LOG_LEVEL.DEBUG, "Starting new session");
-         OneSignalStateSynchronizer.setNewSession();
          if (inForeground) {
+            logger.debug("Starting new session");
+
+            OneSignalStateSynchronizer.setNewSession();
             outcomeEventsController.cleanOutcomes();
             sessionManager.restartSessionIfNeeded(getAppEntryState());
             getInAppMessageController().resetSessionLaunchTime();
+            setLastSessionTime(time.getCurrentTimeMillis());
          }
-      } else if (inForeground) {
-         OneSignal.onesignalLog(LOG_LEVEL.DEBUG, "Continue on same session");
+      } else {
+         logger.debug("Continue on same session");
+
          sessionManager.attemptSessionUpgrade(getAppEntryState());
       }
       getInAppMessageController().initWithCachedInAppMessages();
 
       // We still want register the user to OneSignal if the SDK was initialized
       //   in the background for the first time.
       if (!inForeground && hasUserId())
-         return;
+         logger.debug("doSessionInit on background with already registered user");
 
```

Comment:
The return here was removed was that intentional?

Reply:
yeah, this was preventing the background update